jd4/case.py

```diff
@@ -256,20 +256,54 @@ def read_legacy_cases(config, open):
                           memory_bytes,
                           int(score_str))
 
-def read_yaml_cases(config, open):
-    for case in yaml.safe_load(config)['cases']:
-        if 'judge' not in case:
-            yield DefaultCase(partial(open, case['input']),
-                              partial(open, case['output']),
-                              parse_time_ns(case['time']),
-                              parse_memory_bytes(case['memory']),
-                              int(case['score']))
-        else:
-            yield CustomJudgeCase(partial(open, case['input']),
+def read_auto_cases(open, zip_file, time_limit='1s', memory_limit='128m'):
+    cnt = 0
+    files = zip_file.namelist()
+    prefix = ''
+    cases = []
+    re_input = re.compile(r'([a-zA-Z]*)([0-9]*)\.in')
+    for i in files:
+        match = re_input.match(i)
+        if match:
+            prefix = match.group(1)
+            cases.append(int(match.group(2)))
+            cnt += 1
+    if not cnt:
+        raise FormatError('No test case found.')
+    cases.sort()
+    counter = 0
+    score = 100 // cnt
+    divider = cnt - 100%cnt
+    for i in cases:
```

Comment:
See the documentation of `enumerate`, and consider using `i, case in enumerate(cases)`.